api/controllers/v0.2/createPatron.js

```diff
@@ -55,74 +55,155 @@ function renderResponse(req, res, status, message) {
 }
 
 /**
- * createPatron(req, res)
- * The callback for the route "/patrons".
- * It will fire a POST request to the ILS API for creating a new patron.
+ * streamPatron(req, res, streamPatron, modeledResponse)
+ * Stream the new patron data to Kinesis
  *
  * @param {HTTP request} req
  * @param {HTTP response} res
+ * @param {object} streamPatron
+ * @param {object} modeledResponse
  */
-function createPatron(req, res) {
-  if (!process.env.ILS_CREATE_TOKEN_URL || isEmpty(process.env.ILS_CREATE_TOKEN_URL)) {
-    res
-      .status(500)
-      .header('Content-Type', 'application/json')
-      .json(modelResponse.errorResponseData(
-        collectErrorResponseData(
-          null,
-          'invalid-request',
-          'The create-patron service is missing the ILS_CREATE_TOKEN_URL.',
-          null,
-          { form: ['The create-patron service is misconfigured.'] } // eslint-disable-line comma-dangle
-        ) // eslint-disable-line comma-dangle
-      ));
+function streamPatron(req, res, streamPatronData, modeledResponse) {
+  streamPublish.streamPublish(
+    process.env.PATRON_SCHEMA_NAME_V02,
+    process.env.PATRON_STREAM_NAME_V02,
+    streamPatronData // eslint-disable-line comma-dangle
+  ).then((streamResponse) => { // eslint-disable-line no-unused-vars
+    renderResponse(req, res, 201, modeledResponse);
+    console.log('Published to stream successfully!'); // eslint-disable-line no-console
+  }).catch((streamError) => {
+    renderResponse(req, res, 201, streamError);
+    console.error(`Error publishing to stream: ${streamError}`); // eslint-disable-line no-console
```

Comment:
Is this the right http status code for the condition where the patron was created but there was an error writing to the stream? Maybe it is; I'm not sure. But it may be confusing to send a different payload for the same 201 status code - `streamError` instead of `modeledResponse`

Also, Is this an opportunity to centralize error handling by returning the promise without a `catch` call so that the caller has to handle the error? (i.e. allow it to be caught by the `catch` on what is now line 104).